Content.Client/Administration/UI/Notes/AdminNotesLine.xaml.cs

```diff
@@ -1,85 +1,92 @@
+using Robust.Client.Graphics;
+using Robust.Client.ResourceManagement;
 using Robust.Client.UserInterface;
 using Robust.Client.UserInterface.Controls;
 using Robust.Client.UserInterface.XAML;
 using Robust.Shared.Input;
-using static Robust.Client.UserInterface.Controls.LineEdit;
+using Robust.Shared.Utility;
 
 namespace Content.Client.Administration.UI.Notes;
 
 [GenerateTypedNameReferences]
 public sealed partial class AdminNotesLine : BoxContainer
 {
-    private RichTextLabel? _label;
-    private LineEdit? _edit;
+    private readonly SpriteSystem _sprites;
 
-    public AdminNotesLine(SharedAdminNote note)
+    private const string AdminNotesTextureBase = "/Textures/Interface/AdminNotes/";
+    private static readonly Dictionary<NoteSeverity, string> SeverityIcons = new()
     {
+        { NoteSeverity.None, AdminNotesTextureBase + "none_button.png" },
+        { NoteSeverity.Minor, AdminNotesTextureBase + "minor_button.png" },
+        { NoteSeverity.Medium, AdminNotesTextureBase + "medium_button.png" },
+        { NoteSeverity.High, AdminNotesTextureBase + "high_button.png" },
+    };
+    private static readonly Dictionary<NoteType, string> NoteTypeIcons = new()
+    {
+        { NoteType.Message, AdminNotesTextureBase + "message.png" },
+        { NoteType.Watchlist, AdminNotesTextureBase + "watchlist.png" },
+    };
+
+    public AdminNotesLine(SpriteSystem sprites, SharedAdminNote note)
+    {
+        _sprites = sprites;
         RobustXamlLoader.Load(this);
 
         Note = note;
         MouseFilter = MouseFilterMode.Pass;
 
-        AddLabel();
+        Separator.Visible = true;
+        Refresh();
     }
 
     public SharedAdminNote Note { get; private set; }
     public int Id => Note.Id;
-    public string OriginalMessage => Note.Message;
-    public string EditText => _edit?.Text ?? OriginalMessage;
 
-    public event Action<AdminNotesLine>? OnSubmitted;
     public event Func<AdminNotesLine, bool>? OnClicked;
 
-    private void AddLabel()
+    /// <summary>
+    /// Attempts to refresh the current note line with new data. The note it draws data on is stored in <see cref="Note"/>
+    /// </summary>
+    private void Refresh()
     {
-        if (_edit != null)
+        try
         {
-            _edit.OnTextEntered -= Submitted;
-            _edit.OnFocusExit -= Submitted;
-
-            RemoveChild(_edit);
-            _edit = null;
+            SeverityRect.Texture = _sprites.Frame0(new SpriteSpecifier.Texture(new ResourcePath(
+                Note.NoteType == NoteType.Note
+                    ? SeverityIcons[Note.NoteSeverity]
+                    : NoteTypeIcons[Note.NoteType]
+            )));
         }
-
-        _label = new RichTextLabel();
-        _label.SetMessage(Note.Message);
-
-        AddChild(_label);
-        _label.SetPositionFirst();
-
-        Separator.Visible = true;
-    }
-
-    private void AddLineEdit()
-    {
-        if (_label != null)
+        catch (KeyNotFoundException)
```

Comment:
TryGetValue

Reply:
I am aware, but instead of doing two try get values I opted for a try-catch

Reply:
> I am aware, but instead of doing two try get values I opted for a try-catch

Absolutely never use try catch unless it's required.

Reply:
I think the try catch version was clearer to read, but implemented the changes anyway.